lib/active_merchant/billing/gateways/decidir_plus.rb

```diff
@@ -106,10 +135,17 @@ def add_fraud_detection(post, options)
         return unless fraud_detection = options[:fraud_detection]
 
         {}.tap do |hsh|
-          hsh[:send_to_cs] = fraud_detection[:send_to_cs] ? true : false # true/false
+          hsh[:send_to_cs] = fraud_detection[:send_to_cs] == 'true' # true/false
           hsh[:channel] = fraud_detection[:channel] if fraud_detection[:channel]
           hsh[:dispatch_method] = fraud_detection[:dispatch_method] if fraud_detection[:dispatch_method]
-          hsh[:csmdds] = fraud_detection[:csmdds] if fraud_detection[:csmdds]
+          csmdds_arr = []
+          fraud_detection[:csmdds].each do |csmdds|
+            csmdds_hsh = {}
+            csmdds_hsh[:code] = csmdds[:code].to_i
+            csmdds_hsh[:description] = csmdds[:description]
+            csmdds_arr.append(csmdds_hsh)
+          end
```

Comment:
This code will result in an exception if `:csmdds` isn't set in the `options[:fraud_detection]` hash. I think we should break this code out into it's own method and you can call it with a conditional, `if fraud_detection[:csmdds]` or return early from the method.

Reply:
Thanks, change is updated with unit test case as well.